### PR TITLE
scipy: upstream Linux fix

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/scipy.rb [hunk 1]
```diff
@@ -46,7 +48,7 @@ def install
     version = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
     ENV["PYTHONPATH"] = Formula["numpy"].opt_lib/"python#{version}/site-packages"
     ENV.prepend_create_path "PYTHONPATH", lib/"python#{version}/site-packages"
-    system Formula["python@3.9"].opt_bin/"python3", "setup.py", "build", "--fcompiler=gnu95"
+    system Formula["python@3.9"].opt_bin/"python3", "setup.py", "build", "--fcompiler=gfortran"
```
Comment (carlocab) [inline Formula/scipy.rb:51]: This depends on `numpy`, though, which also specifies `--fcompiler=gnu95`. Doesn't that need to be fixed first?

Comment (danielnachun) [reply Formula/scipy.rb:51]: Yes and oddly I see that on Linux this was not changed to `gfortran`, while it was for `scipy`.  I'm not sure why this isn't consistent, but I believe `gfortran` is the better choice for both.  I will change this in `numpy` first.

Comment (carlocab) [inline Formula/scipy.rb:51]:
```suggestion
    system Formula["python@3.9"].opt_bin/"python3", "setup.py", "build", "--fcompiler=gfortran", "--parallel=#{ENV.make_jobs}"
```

**Comment by danielnachun** [comment]
@carlocab would mind double checking if this looks good now?  The `numpy` changes are merged now.

File: Formula/scipy.rb (hunk 1 above)
Comment (carlocab) [reply Formula/scipy.rb:51]: You can probably add this.

**Review by carlocab — DISMISSED** [review]

**Comment by danielnachun** [comment]
I had to update this because so that it had a signed commit, but the changes are already approved so I will merge.

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
Squash-and-merge would've worked too even if the original commit wasn't signed.

**Pull request merged**